src/MediaWiki/Search/SearchEngineFactory.php

```diff
@@ -22,31 +23,38 @@ class SearchEngineFactory {
 	/**
 	 * @since 3.1
 	 *
-	 * @param IDatabase $connection
+	 * @param mixed $connection Either IConnectionProvider (MW 1.41+) or IDatabase (MW 1.40)
 	 *
 	 * @return SearchEngine
 	 * @throws SearchEngineInvalidTypeException
 	 */
-	public function newFallbackSearchEngine( IDatabase $connection = null ) {
+	public function newFallbackSearchEngine( $connection = null ) {
 		$applicationFactory = ApplicationFactory::getInstance();
 		$settings = $applicationFactory->getSettings();
 
 		if ( $connection === null ) {
+			// For MW 1.41+, getConnectionManager()->getConnection() returns IConnectionProvider
+			// For MW 1.40, it returns IDatabase
 			$connection = $applicationFactory->getConnectionManager()->getConnection( DB_REPLICA );
```

Comment:
I don’t see this changed in getConnection for MW1.41+.

Reply:
@paladox Yes, this needs to be updated, that is something I am working on. 
One test is failing regarding this, and that test is skipped at the moment. 
I am sure it will work as soon as we get the right connection for MW 1.41.

If you have any idea, please, share it here :)

Reply:
What’s the failure?

I’m not sure how to fix this in getConnection but I presume you do / or is trying. I hadn’t try to figure out how to fix the test until the main part was fixed.

Reply:
Use getDBLoadBalancerFactory.

Reply:
See https://github.com/wikimedia/mediawiki/commit/5eae1de53f8b63cfe960109511636603fb7f804d#diff-2561445758ddc512c14a769dbc75e9251af0f4d99356218e36126ed44db09b73R1860